### ISSUE TITLE
Error raised when analyzing varargs method call

### ISSUE THREAD

**Issue by FlavienQui** [integrated-application-development/sonar-delphi#297]
### Prerequisites

- [X] This bug is in SonarDelphi, not SonarQube or my Delphi code.
- [X] This bug has not already been reported.

### SonarDelphi version

1.6.0

### Issue description

Running analysis with DelphiLint 1.1.1 in Standalone mode, an error is raised when the code contains a method call with the varargs directive.
![Capture](https://github.com/user-attachments/assets/495ff1d0-d69d-4160-9d64-206c5d8cb109)

### Steps to reproduce

Analyze the file with the minimal Delphi code joined to this issue.

### Minimal Delphi code exhibiting the issue

```
unit Unit1;

interface

procedure MyProcedure;

var
  SetValues: procedure(AValues: Integer); cdecl varargs;

implementation

procedure MyProcedure;
var
  LValue: Integer;
begin
  LValue := 1;
  SetValues(LValue, LValue);
end;

end.
```

**Comment by Cirras** [comment]
Hi @FlavienQui,

The exception is getting raised from the `VariableInitialization` rule, but the actual problem goes a bit deeper into the analyzer's semantic model. It looks like we don't consistently persist `varargs` when generating type information for procedurals.

Thanks for reporting this bug, your minimal example was very helpful.
